Core/include/Acts/Geometry/GeometryID.hpp

```diff
@@ -109,6 +113,10 @@ class GeometryID {
   friend constexpr bool operator<(GeometryID lhs, GeometryID rhs) {
     return lhs.m_value < rhs.m_value;
   }
+
+  // provide access access to the masks
+  template <typename element_t>
+  friend class HierarchicalGeometryContainer;
```

Comment:
Can we not just make the masks public?

Reply:
This was discussed already in the context of the [GeometryID interface streamline](https://gitlab.cern.ch/acts/acts-core/-/merge_requests/693). The exact masks are an implementation detail that is not used anywhere in the code. Everything uses the GeometryID accessors only.

This container is the only exception: using the mask directly allows for simpler implementations compared to using only the accessors. I think that is exactly the use-case for `friends`.

Reply:
`friends` is a pretty bad code smell, and should be avoided if at all possible I believe.

Reply:
I disagree. Overuse of `friend`, yes. But this is exactly the use-case for it: two components that internally use private information to significantly simplify the implementation and that in both cases is not exported.

If you change e.g. the order of levels in the geometry id than you would also have to adapt the container. This is by design and can not be avoided. So the two components are already coupled. Adding `friend` here does not increase the coupling beyond what is already there.

Reply:
Ok, I disagree, but let's leave it at that.

Reply:
I looked in a bit more detail at the code and I think I can still have a reasonable implementation without the `friend`. If I have something by tomorrow, I will switch.

Reply:
The implementation now uses only the available `GeometryID` accessors. The `friend` declaration has been removed. I was wrong about the implementation being more complicated.

Reply:
Cool